.drone.starlark

```diff
@@ -0,0 +1,950 @@
+										]
+									}
+								] if federatedServerNeeded else []) + [
+									setupServerAndApp(phpVersion),
+									installExtraApps(phpVersion, extraApps),
+									extraSetup,
+									fixPermissions(phpVersion),
+									({
+										'name': 'acceptance-tests',
+										'image': 'owncloudci/php:%s' % phpVersion,
+										'pull': 'always',
+										'environment': environment,
+										'commands': [
+											'touch /var/www/owncloud/saved-settings.sh',
+											'. /var/www/owncloud/saved-settings.sh',
+											'cd /var/www/owncloud/testrunner/apps/%s' % config['app'],
+											'make test-acceptance-webui'
+										]
+									} if isWebUI else None),
+									({
+										'name': 'acceptance-tests',
+										'image': 'owncloudci/php:%s' % phpVersion,
+										'pull': 'always',
+										'environment': environment,
+										'commands': [
+											'cd /var/www/owncloud/testrunner/apps/%s' % config['app'],
+											'make test-acceptance-api'
+										]
+									} if isAPI else None),
+									({
+										'name': 'acceptance-tests',
+										'image': 'owncloudci/php:%s' % phpVersion,
+										'pull': 'always',
+										'environment': environment,
+										'commands': [
+											'cd /var/www/owncloud/testrunner/apps/%s' % config['app'],
+											'make test-acceptance-cli'
+										]
+									} if isCLI else None),
+								],
+								'services': databaseService(db)
+									+ browserService(browser)
+									+ emailService(emailNeeded)
+									+ extraServices
+									+ owncloudService(server, phpVersion, 'server', '/var/www/owncloud/server', False)
+									+ (owncloudService(server, phpVersion, 'federated', '/var/www/owncloud/federated', False) if federatedServerNeeded else []),
+								'depends_on': [],
+								'trigger': {
+									'ref': [
+										'refs/pull/**',
+										'refs/tags/**'
+									]
+								}
+							}
+
+							for branch in config['branches']:
+								result['trigger']['ref'].append('refs/heads/%s' % branch)
+
+							pipelines.append(result)
+
+	if errorFound:
+		return False
+
+	return pipelines
+
+def notify():
+	result = {
+		'kind': 'pipeline',
+		'type': 'docker',
+		'name': 'chat-notifications',
+		'clone': {
+			'disable': True
+		},
+		'steps': [
+			{
+				'name': 'notify-rocketchat',
+				'image': 'plugins/slack:1',
+				'pull': 'always',
+				'settings': {
+					'webhook': {
+						'from_secret': 'public_rocketchat'
+					},
+					'channel': config['rocketchat']
+				}
+			}
+		],
+		'depends_on': [],
+		'trigger': {
+			'ref': [
+				'refs/tags/**'
+			]
+		}
+	}
+
+	for branch in config['branches']:
+		result['trigger']['ref'].append('refs/heads/%s' % branch)
+
+	return result
+
+def databaseService(name):
+	if name.startswith('mariadb'):
+		return [{
+			'name': 'mariadb',
+			'image': name,
+			'pull': 'always',
+			'environment': {
+				'MYSQL_USER': 'owncloud',
+				'MYSQL_PASSWORD': 'owncloud',
```

Comment:
the DB credentials are hard-coded here and in `installCore()`

Reply:
done